File: electivehelper-api/src/main/java/cl/ufro/dci/electivehelper_api/domain/Elective.java
package cl.ufro.dci.electivehelper_api.domain;

import java.util.ArrayList;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Elective {
    private String eleCode;
    private String eleName;
    private String eleModule;
    private String eleMode;
    private ArrayList<Slot> eleSlots;

    public void addSlots(ArrayList<Slot> slots) {
        for(Slot slot : slots) {
            this.eleSlots.add(slot);
        }
    }
}

File: electivehelper-ui/src/app/domain/elective.ts
import { Period } from "./period";
import { Slot } from "./slot";

export class Elective{
 eleCode: string;
 eleName: string;
 eleModule: string;
 eleMode: string;
 eleSlots: Slot[];

 constructor(eleCode: string, eleName: string, eleModule: string, eleMode: string, eleSlots: Slot[]){
     this.eleCode = eleCode;
     this.eleName = eleName;
     this.eleModule = eleModule;
     this.eleMode = eleMode;
     this.eleSlots = eleSlots;
 }

 equals(elective: Elective): boolean{
    return this.eleCode === elective.eleCode && this.eleModule === elective.eleModule;
 }

 isInDayPeriod(day: string, period: Period): boolean {
    for (let slot of this.eleSlots) {
      if (slot.sloDay === day && slot.sloPeriod === period.periodName) {
        return true;
      }
    }
    return false;
  }

}